paddle/fluid/operators/fusion_lstm_op.cc

```diff
@@ -270,6 +270,204 @@ class FuisonLSTMKernel : public framework::OpKernel<T> {
   blas.GEMM(CblasNoTrans, CblasNoTrans, bs, D4, D, static_cast<T>(1), prev, D, \
             wh_data, D4, static_cast<T>(1), out, D4)
 
+#ifdef __AVX__
+#define GET_CT_MOVE_ONE_STEP(ct_1, gates, ct)                          \
+  tmp0 = _mm256_mul_ps(_mm256_loadu_ps((const float*)gates + i),       \
+                       _mm256_loadu_ps((const float*)gates + D + i));  \
+  tmp1 = _mm256_mul_ps(_mm256_loadu_ps((const float*)ct_1 + i),        \
+                       _mm256_loadu_ps((const float*)gates + D2 + i)); \
+  tmp0 = _mm256_add_ps(tmp0, tmp1);                                    \
+  _mm256_storeu_ps(reinterpret_cast<float*>(ct) + i, tmp0)
+
+// gates: W_ch, W_ih, W_fh, W_oh
+#define GET_Ct(ct_1, gates, ct)                    \
+  /* C_t = C_t-1 * fgated + cand_gated * igated*/  \
+  act_cand(D, gates, gates);                       \
+  if ((D >= 8) && std::is_same<float, T>::value) { \
+    constexpr int block = 8;                       \
+    const int rest = D % block;                    \
+    const int end = D - rest;                      \
+    int i = 0;                                     \
+                                                   \
+    __m256 tmp0, tmp1;                             \
+    for (i = 0; i < end; i += block) {             \
+      GET_CT_MOVE_ONE_STEP(ct_1, gates, ct);       \
+    }                                              \
+                                                   \
+    if (rest > 0) {                                \
+      i = D - 8;                                   \
+      GET_CT_MOVE_ONE_STEP(ct_1, gates, ct);       \
```

Comment:
I guess, this part is a loop remainder. Shouldn't this part be done sequentially? If `rest` is lower than `block`, you might read some invalid data that follow the buffer you're reading from with `_mm256_loadu_ps` intrinsics.